doc: add minutes for meeting 2 Nov 2022 (#1303)
nodejs/TSC: meetings/2022-11-02.md

# Node.js Technical Steering Committee (TSC) Meeting 2022-11-02

## Links

* **Recording**:  <http://www.youtube.com/watch?v=Rjdl0WvZI_I>
* **GitHub Issue**: <https://github.com/nodejs/TSC/issues/1302>

## Present

* Beth Griggs @BethGriggs (TSC)
* Ruben Bridgewater @BridgeAR (TSC)
* Сковорода Никита Андреевич @ChALkeR (TSC)
* Colin Ihrig @cjihrig (TSC)
* Geoffrey Booth @GeoffreyBooth (TSC)
* Gireesh Punathil @gireeshpunathil (TSC)
* James Snell @jasnell (TSC)
* Joyee Cheung @joyeecheung (TSC)
* Chengzhong Wu @legendecas (TSC)
* Matteo Collina @mcollina (TSC)
* Michael Dawson @mhdawson (TSC)
* Darshan Sen @RaisinTen (TSC)
* Richard Lau @richardlau (TSC)
* Robert Nagy @ronag (TSC)
* Michaël Zasso @targos (TSC)

## Agenda

### Announcements

* Richard

>   * 14.x release yesterday

Reviewer: ```suggestion
  * 14.21.0 release yesterday
```

>   * Security release planned for Thursday Nov 3 across all supported lines, blog post explains

Reviewer: ```suggestion
  * Security release planned for Thursday Nov 3 across all supported lines, [blog post](https://nodejs.org/en/blog/vulnerability/november-2022-security-releases/) explains
```

  * relates to the Version Checker discussion

* Adding a Version Checker [#44942](https://github.com/nodejs/node/issues/44942)
  * Matteo
    * related to encouraging people to update their Node.js version
    * started based on a discussion at NodeConf.eu
    * in favor of adding to Node.js
  * Geoffrey
    * recollection was that there is opposition to building a full fledged update feature
    * asking if there is an example of any runtime tool that has a version checker, answer seems
      to be no. Any existing examples were part of an update feature
    * having this on its own without an update feature without a plan for an update feature
      does not seem like a good idea.
  * Richard, have seen complaints of npm’s version checker/warning.  So not in favor of
    on by default, if opt in ok
  * Darshan, also thinking about the UI.
  * Michael Zasso
    * there are so many different ways to install and update node, most of which are third-party
    * in the current state, we absolutely cannot provide a self-update
    * Michael D, agree on this one, either the check should be agnostic to the install type, or have

>       a #define to remove that other distros can use.

Reviewer 2: ```suggestion
      a `configure` option to remove that other distros can use.
```

  * Richard, only way you can check is to do outbound request, this is one reason why we
    should not do it by default.
  * Nikita, what about only doing in the repl?
    * Geoffrey, I suggested an automatic message when starting the REPL, because that
      would be more certainly a user rather than an automated process running node,
      but people thought it wasn't prominent enough, because people don't run the REPL
      often enough
    * Nikita, maybe we can just check against schedule and warn if major version is no longer supported
      based on schedule
  * Joyee
    * Sending outbound request by default sounds quite annoying…we shouldn’t assume
      that users have stable/working connections by default
  * Matteo, we could check when version is LTS, could warn when that occurs, does not required
    any output request.
    * Geoffrey would that break people?
    * Matteo, only print if there is a TTY attached.
  * Ruben, may want to have a user configuration file in general, as possibly it should be
    a project that gets notification, not an individual user. In the repl, possibly could warn

>     6 months,  

Reviewer 2: ```suggestion
    6 months,
```

  * Geoffrey - have opened a issue about a .env file for Node.js
    * Colin might want more than a single file, possibly a directory
  * Matteo, part of the problem is that we don’t have a good recommendation for how
    people install Node.js. A good outcome of the discussion would be that although
    we won’t ship a self-updater because there are good ones in the community
  * Richard, there is a page, discovery may be a challenge, it does list the package
    Managers.
    * <https://nodejs.org/en/download/package-manager/>
  * Matteo, from discussion, feature on it’s own might not be enough on its own, but is

>     a good area tp discuss

Reviewer 3: ```suggestion
    a good area to discuss
```
